### PR TITLE
REV-022-1: create FormFileInput shared component

### PR DESCRIPTION
**Pull request by hosamAjlouni-op**
Implemented a shared Form File Input component.
Implemented a shared File List component.
Implemented a Listing Image File Input component for (create listing page).

figma: [Form File Input](https://www.figma.com/file/22d3bKcFL0ipWIZJOUY3Km/reverifi?node-id=0%3A75041)

### CODE REVIEW

File: src/components/shared/FormFileInput/index.jsx [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+  InputSection,
+  InputSpace,
+  SelectedItem,
+  SelectedItemInnerSection,
+  SelectedItemsList,
+  ThumbnailWrapper,
+  UploadIcon,
+} from './formFileInput.styles';
+
+const types = {
```
Comment (AhmadAlShwaiki) [inline src/components/shared/FormFileInput/index.jsx:19]:
```suggestion
const allowedFileTypes = {
```

**Review by mohdabed123 — APPROVED** [review]

File: src/components/create-listing/ListingImageInput/index.jsx [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  UploadIcon,
+} from './listingImageInput.styles';
+
+function ListingImageInput({ files, onAddImages, onDeleteImage }) {
+  return (
+    <Container>
+      <h2>Upload Images</h2>
+      <InnerContainer>
+        <ImageInputSection>
+          <FormFileInput
```
Comment (AhmadAlShwaiki) [inline src/components/create-listing/ListingImageInput/index.jsx:19]:
```suggestion
          <FileInput
```

File: src/components/create-listing/ListingImageInput/index.jsx [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+  InnerContainer,
+  InputInterface,
+  UploadIcon,
+} from './listingImageInput.styles';
+
+function ListingImageInput({ files, onAddImages, onDeleteImage }) {
+  return (
+    <Container>
+      <h2>Upload Images</h2>
+      <InnerContainer>
```
Comment (AhmadAlShwaiki) [inline src/components/create-listing/ListingImageInput/index.jsx:17]:
```suggestion
      <Wrapper>
```

File: src/components/create-listing/ListingImageInput/index.jsx [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+            <InputInterface>
+              <UploadIcon />
+              <span>Drag and drop here</span>
+              <span>or</span>
+              <span>browse</span>
+            </InputInterface>
+          </FormFileInput>
+        </ImageInputSection>
+
+        <ImageInputSection>
```
Comment (AhmadAlShwaiki) [inline src/components/create-listing/ListingImageInput/index.jsx:33]: Is this component needed?
```suggestion
```

File: src/components/shared/FormFileInput/index.jsx [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+import PropTypes from 'prop-types';
+import { Input, InputWrapper } from './formFileInput.styles';
+
+/**
+ * Renders a form file input
+ *
+ * @param  {string}   acceptedType  Accepted files types.
+ * @param  {boolean}  multiple      Determines files quantity to single or many.
+ * @param  {Function} setFiles      A function to set and update files.
+ *
```
Comment (AhmadAlShwaiki) [inline src/components/shared/FormFileInput/index.jsx:11]:
```suggestion
 @param {children}
 *
```

File: src/components/create-listing/ListingImageInput/index.jsx (hunk 4 above)
Comment (hosamAjlouni-op) [reply src/components/create-listing/ListingImageInput/index.jsx:33]: yes for sure

File: src/components/create-listing/ListingImageInput/index.jsx (hunk 2 above)
Comment (hosamAjlouni-op) [reply src/components/create-listing/ListingImageInput/index.jsx:19]: I think that this is consistent with other shared form inputs, except if you want me to change it where it has been imported only.

File: src/components/create-listing/ListingImageInput/index.jsx (hunk 3 above)
Comment (hosamAjlouni-op) [reply src/components/create-listing/ListingImageInput/index.jsx:17]: done.

File: src/components/shared/FormFileInput/index.jsx (hunk 5 above)
Comment (hosamAjlouni-op) [reply src/components/shared/FormFileInput/index.jsx:11]: done.

File: src/components/create-listing/ListingImageInput/index.jsx (hunk 4 above)
Comment (hosamAjlouni-op) [reply src/components/create-listing/ListingImageInput/index.jsx:33]: this handles the size of the flex items.

**Review by AhmadAlShwaiki — APPROVED** [review]
LGTM

**Pull request merged**